### PR TITLE
Enhanced install.sh to optionally authenticate user

### PR DESCRIPTION
**Pull request by eBardX**

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -11,32 +11,59 @@ waldo_exec2_name="sim_appcenter_build_and_upload.sh"
...
 waldo_asset2_name=
 
+waldo_exec_path=
+
 waldo_found_in_path=false
 waldo_is_reinstall=false
 
+function authorize() {
+    local _user_token=${WALDO_USER_TOKEN:-${TOKEN:-}}
+    local _auth_done=false
```
Comment (vbfox) [inline install.sh:21]: Unused variable ?

File: install.sh [hunk 2]
```diff
@@ -11,32 +11,59 @@ waldo_exec2_name="sim_appcenter_build_and_upload.sh"
...
+        do_authorization $_user_token
+
+        _auth_status=$?
+    else
+        _auth_status=1
+    fi
+
+    if (( $_auth_status != 0 )); then
+        echo ""
+        echo "You are not currently authorized to use most Waldo CLI commands."
```
Comment (vbfox) [inline install.sh:38]: authorized sounds a bit like 👮🏼‍♂️ you're not allowed no ? Maybe authenticated ?

**Review by vbfox — APPROVED** [review]

File: install.sh (hunk 1 above)
Comment (eBardX) [reply install.sh:21]: Good catch. A remnant of an earlier iteration.

**Review by VincentJaouen — APPROVED** [review]
LGTM

File: install.sh (hunk 2 above)
Comment (eBardX) [reply install.sh:38]: Perhaps. “authorize” and “authenticate” actually have 2 distinct meanings in the world of computer security (as you are no doubt aware) — and I always get them confused. I need to go re-read the definitions to convince myself that you are correct. Trust but verify.

Comment (eBardX) [reply install.sh:38]: You are correct. `waldo auth` is definitely _authenticating_. I will need to update the help for that command in future pass. I will adjust the install message accordingly.

**Pull request merged**